### ISSUE TITLE
Embedded arrays or nested arrays

### ISSUE THREAD

**Issue by Ractivuss** [alonrbar/easy-template-x#36]
Hi! I'm having some trouble with some nested arrays with objects. In my document I do this:
`{#array}
   {property1} {property2}
   {#nestedArray}
     {propertyOneOfNestedArray}
   {/nestedArray}
{/array}`

Obviously this is an example, but this is some kind of example that I'm trying to do.

Sorry for my english.

**Comment by alonrbar** [comment]
After looking into it a bit it does seem like a bug...
It looks like the bug is in the [LoopParagraphStrategy.splitBefore](https://github.com/alonrbar/easy-template-x/blob/master/src/plugins/loop/strategy/loopParagraphStrategy.ts) method but to find the exact cause and fix it will take a little longer.
In the meanwhile, if it's relevant in your case you can split this line into separate paragraphs to get a valid result, like so:

```
{#arrray} 
{property1} {property2} 
{#nestedArray} 
{propertyOneOfNestedArray} 
{/nestedArray} 
{/arrray}
```

**Comment by alonrbar** [comment]
Also worth mentioning, PRs are always welcome 😄

**Comment by Ractivuss** [comment]
That worked! Thanks!

**Comment by alonrbar** [comment]
Fixed in v1.0.0
